### PR TITLE
Have Frame use MinimumHeight/Width for minimums instead of constraints

### LINKED ISSUE #13074: FlexLayout not working after update
### Description

Prior to upgrading VS2022 Preview yesterday, the FlexLayout implemented in my code was working 100%. Since the changes, the FlexLayout does not render properly. Instead of displaying like a grid, it is now overlapped and displays in rows.

### Steps to Reproduce

Create a new MAUI project and target only Android 13 (v33).
Replace the ContentPage with the exact code I tested with:

```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="MauiTestBench.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml">


 <ContentPage.Resources>
        <ResourceDictionary>
            <!--  Start Tile Styling  -->
            <Style x:Key="TileStyle" TargetType="Frame">
                <Setter Property="Padding" Value="0,0,0,0" />
                <Setter Property="Margin" Value="0,20,0,20" />
                <Setter Property="BorderColor" Value="LightBlue" />
                <Setter Property="CornerRadius" Value="15" />
                <Setter Property="HasShadow" Value="True" />
                <Setter Property="HeightRequest" Value="120" />
                <Setter Property="WidthRequest" Value="120" />
                <Setter Property="MinimumHeightRequest" Value="100" />
                <Setter Property="MinimumWidthRequest" Value="100" />
                <Setter Property="HorizontalOptions" Value="CenterAndExpand" />
                <Setter Property="VerticalOptions" Value="CenterAndExpand" />
            </Style>
            <Style x:Key="TileImageStyle" TargetType="Image">
                <Setter Property="HeightRequest" Value="50" />
                <Setter Property="WidthRequest" Value="50" />
                <Setter Property="Margin" Value="5" />
            </Style>
            <Style x:Key="TileLabelStyle" TargetType="Label">
                <Setter Property="FontAutoScalingEnabled" Value="True" />
                <Setter Property="FontSize" Value="Default" />
                <Setter Property="HorizontalTextAlignment" Value="Center" />
                <Setter Property="TextColor" Value="Black" />
                <Setter Property="TextTransform" Value="Uppercase" />
                <Setter Property="FontFamily" Value="QuicksandRegular" />
                <Setter Property="FontSize" Value="Small" />
            </Style>
            <!--  End Tile Styling  -->
        </ResourceDictionary>
    </ContentPage.Resources>
    <ScrollView>
        <FlexLayout
            x:Name="FlexTiles"
            AlignItems="Stretch"
            Direction="Row"
            JustifyContent="SpaceEvenly"
            VerticalOptions="Start"
            Wrap="Wrap">

            <!--  ROSTERS TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  ADVANCED TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  MY HOLIDAYS TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  EMPLOYEE PAY TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  RUNSHEET TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  SOS TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  SMARTFORMS TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  MY MESSAGES TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  EMAIL CONTROL ROOM TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  CALL CONTROL ROOM TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>

            <!--  LOGOFF TILE  -->
            <Frame Style="{DynamicResource TileStyle}">
                <StackLayout HorizontalOptions="Center" VerticalOptions="Center">
                    <Image Style="{DynamicResource TileImageStyle}" />
                    <Label Style="{DynamicResource TileLabelStyle}" />
                </StackLayout>
            </Frame>
        </FlexLayout>
    </ScrollView>

</ContentPage>
```

Run the demo and you should see the above frames are displayed in a stacked format.

This has been tested on VS2022 17.4.4 and the Preview version is 17.5

### Link to public reproduction project repository

n/a

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android

### Affected platform versions

Android 13

### Did you find any workaround?

No workaround, so I implemented a grid until this gets fixed.

### Relevant log output

_No response_

### LINKED ISSUE #13469: Frame Error display on Android devices
### Description

image
image

I use a Frame -Control with image to display an avatar. On iOS everything works as expected (Image1). On Android, the dimensions of the frame do not fit (Image2). 
_The really funny thing is that when using a Border control, the behavior is the other way around. Border works on Android but not on iOS._

`<Frame 
                            BorderColor="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource Gray950}}"
                            CornerRadius="25"
                            HeightRequest="50"
                            WidthRequest="50"
                            IsClippedToBounds="True"
                            HorizontalOptions="Start"
                            VerticalOptions="Start">
                                <Image Source="{Binding UserProfileImage}" VerticalOptions="Center" HorizontalOptions="Center"
                              Aspect="AspectFill"
                              HeightRequest="50"
                              WidthRequest="50">
                                    <Image.GestureRecognizers>
                                        <TapGestureRecognizer Tapped="TapGestureRecognizer_PowerProfile" CommandParameter="{Binding UId}"   NumberOfTapsRequired="1" />
                                    </Image.GestureRecognizers>
                                </Image>
                            </Frame>`

### Steps to Reproduce

...

### Link to public reproduction project repository

...

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 13

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

The legacy FrameRenderer on Android is using the measurement constraints as minimum values, which means all Frames attempt to consume the size they're measured with.

These changes use the MinimumHeight and MinimumWidth as the minimums for the Frame, and add tests for this scenario.

### Issues Fixed

Fixes #13074
Fixes #13469

### CODE REVIEW

**Comment by rmarinho** [comment]
Failing tests on Controls _> Android

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs [hunk 1]
```diff
@@ -96,17 +97,28 @@ public FrameRenderer(Context context, IPropertyMapper mapper, CommandMapper comm
...
+		Size IViewHandler.GetDesiredSize(double widthConstraint, double heightConstraint)
 		{
-			double minWidth = 20;
-			if (Primitives.Dimension.IsExplicitSet(widthMeasureSpec) && !double.IsInfinity(widthMeasureSpec))
-				minWidth = widthMeasureSpec;
+			var virtualView = (this as IViewHandler).VirtualView;
+			if (virtualView == null)
+			{
+				return Size.Zero;
+			}
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:106]:
```suggestion
			var virtualView = (this as IViewHandler)?.VirtualView;
			if (virtualView is null)
			{
				return Size.Zero;
			}
```
As can return a null pointer, so VirtualView could be called in a null ptr and get a NRE

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

**Review by mandel-macaque — APPROVED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs [hunk 2]
```diff
@@ -96,17 +97,28 @@ public FrameRenderer(Context context, IPropertyMapper mapper, CommandMapper comm
...
 
-		Size IViewHandler.GetDesiredSize(double widthMeasureSpec, double heightMeasureSpec)
+		Size IViewHandler.GetDesiredSize(double widthConstraint, double heightConstraint)
 		{
-			double minWidth = 20;
-			if (Primitives.Dimension.IsExplicitSet(widthMeasureSpec) && !double.IsInfinity(widthMeasureSpec))
-				minWidth = widthMeasureSpec;
+			var virtualView = (this as IViewHandler)?.VirtualView;
+			if (virtualView is null)
+			{
```
Comment (rmarinho) [inline src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:104]: Do we have a way to test this also?

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs [hunk 3]
```diff
@@ -96,17 +97,28 @@ public FrameRenderer(Context context, IPropertyMapper mapper, CommandMapper comm
...
+				minWidth = 20; // Legacy minimum value
+			}
 
-			double minHeight = 20;
-			if (Primitives.Dimension.IsExplicitSet(widthMeasureSpec) && !double.IsInfinity(heightMeasureSpec))
-				minHeight = heightMeasureSpec;
+			if (!Primitives.Dimension.IsExplicitSet(minHeight))
+			{
+				minHeight = 20; // Legacy minimum value
```
Comment (rmarinho) [inline src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:118]: can this be more explicit as a constant?

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs (hunk 2 above)
Comment (hartez) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:104]: There's a test in this PR.

Comment (hartez) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:104]: Unless you mean the `null` scenario?

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs (hunk 3 above)
Comment (hartez) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs]: Done.

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs (hunk 2 above)
Comment (rmarinho) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:105]: Yeah when would it get Size zero?

File: src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.Tests.cs [hunk 4]
```diff
@@ -287,6 +287,14 @@ public async Task ReturnsNonEmptyNativeBoundingBounds(int size)
 			{
 				// https://github.com/dotnet/maui/issues/11020
 			}
+
+			// This type name check is gross, but this project doesn't know anything about Frame/FrameStub
```
Comment (PureWeen) [inline src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.Tests.cs:291]: The frame tests originate from here `FrameHandlerTest.Android.cs` so you could override this test there to provide context in a less-ish gross way

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs (hunk 2 above)
Comment (hartez) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:105]: If someone disconnect the handler in a non-UI thread, or something weird like that. The point is that if the VirtualView is null, then having any size at all doesn't make sense.

File: src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.Tests.cs (hunk 4 above)
Comment (hartez) [reply src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.Tests.cs]: Updated.

**Review by PureWeen — APPROVED** [review]

**Pull request merged**